Simplenote/Classes/SPEditorTextView.m

```diff
@@ -414,4 +423,92 @@ - (BOOL)isNewVerticalMovementForPosition:(UITextPosition *)position inDirection:
     return newMovement;
 }
 
+#pragma mark checklists
+- (void)processChecklists {
+    if (self.attributedText.length == 0) {
+        return;
+    }
+    
+    [self.textStorage setAttributedString:[NSAttributedString attributedStringWithChecklistAttachments: self.attributedText withColor:[theme colorForKey:@"textColor"]]];
+}
+
+// Processes content of note editor, and replaces special string attachments with their plain
+// text counterparts. Currently supports markdown checklists.
+- (NSString *)getPlainTextContent {
+    NSMutableAttributedString *adjustedString = [[NSMutableAttributedString alloc] initWithAttributedString:self.attributedText];
+    // Replace checkbox images with their markdown syntax equivalent
+    [adjustedString enumerateAttribute:NSAttachmentAttributeName inRange:[adjustedString.string rangeOfString:adjustedString.string] options:0 usingBlock:^(id  _Nullable value, NSRange range, BOOL * _Nonnull stop) {
+        if ([value isKindOfClass:[SPTextAttachment class]]) {
+            SPTextAttachment *attachment = (SPTextAttachment *)value;
+            NSString *checkboxMarkdown = attachment.isChecked ? MarkdownChecked : MarkdownUnchecked;
+            [adjustedString replaceCharactersInRange:range withString:checkboxMarkdown];
+        }
+    }];
+    
+    return adjustedString.string;
+}
+
+- (void)insertNewChecklist {
+    NSMutableAttributedString *newString = [[NSMutableAttributedString alloc] initWithAttributedString:self.attributedText];
+    [newString appendAttributedString:[[NSAttributedString alloc] initWithString:@"- [ ] "]];
+    NSString *checkboxText = [MarkdownUnchecked stringByAppendingString:@" "];
+    if (self.selectedRange.location > 0) {
+        checkboxText = [@"\n" stringByAppendingString:checkboxText];
+    }
+    
+    NSTextStorage *storage = self.textStorage;
+    [storage beginEditing];
+    [storage replaceCharactersInRange:self.selectedRange withString:checkboxText];
+    [storage endEditing];
+    
+    // Update the cursor position
+    [self setSelectedRange:NSMakeRange(self.selectedRange.location + 3, self.selectedRange.length)];
```

Comment:
(Proooobably) a good idea to replace the `+ 3` with a calculated value (OR) constant?